sycl/source/detail/pi_opencl.cpp

```diff
@@ -74,8 +77,50 @@ pi_result OCL(piextDeviceSelectBinary)(
   // where context->dispatch is set to the dispatch table provided by PI
   // plugin for platform/device the ctx was created for.
 
-  *selected_image = num_images > 0 ? images[0] : nullptr;
-  return PI_SUCCESS;
+  // Choose the binary target for the provided device
+  const char *image_target;
+  // Get the type of the device
+  cl_device_type device_type;
+  cl_int ret_err = clGetDeviceInfo(cast<cl_device_id>(device), CL_DEVICE_TYPE,
+                                   sizeof(cl_device_type), &device_type, NULL);
+  if (ret_err != CL_SUCCESS) {
+    *selected_image = nullptr;
+    return cast<pi_result>(ret_err);
+  }
+
+  switch (device_type) {
+  // TODO: Factor out vendor specifics into a separate interface
```

Comment:
The pi_opencl.cpp will soon become a standalone plugin libpi_opencl.so. Do you suggest that we have different plugins for different vendors, e.g. libpi_opencl_intel.so and libpi_opencl_amd.so? What is the benefit of this?

Reply:
If vendor-neutrality of this source is not important, I'm fine with dropping this. @kbobrovs, what are your thoughts on this?

Reply:
@smaslov-intel, the idea is not to have plugins per OpenCL vendors, just refactoring the code so that vendor specific stuff goes into separate sources and/or directories, and this source is kept vendor-neutral.
@AGindinson , just change "interface" to "sources"

Reply:
I can see some benefit in some vendor-specific source-code separation, but in this case it really looks like not being worth it. Anyway, I am OK with the todo note.

Reply:
Once devices from more vendors get supported with AOT, they may have their own logic for selecting "best" binary, specific environment variables, etc. Putting all this in one place and maintaining will be impractical.

Reply:
Comment fixed